src/bsoncxx/builder/list/document.hpp

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/config/prelude.hpp>
+
+#include <bsoncxx/builder/core.hpp>
+#include <bsoncxx/exception/error_code.hpp>
+#include <bsoncxx/exception/exception.hpp>
+#include <bsoncxx/types/bson_value/value.hpp>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+namespace builder {
+namespace list {
+using namespace bsoncxx::types;
+
+///
+/// A JSON-like builder for creating BSON documents.
+///
+class document {
+   public:
+    ///
+    /// Default constructor. Creates an empty document.
```

Comment:
A bit of a nit, but it's a little confusing referring to this class as a "document".  Technically it is `class document`, but logically it's a document _builder_.  I think it would be more clear for users if these comments referred to this class as a document builder, ie "Creates an empty document builder", rather than just a document, so that's it's extra clear we're not talking about a `bsoncxx::document`.